### PR TITLE
Add aggregate read option

### PR DESCRIPTION
**Pull request by Licenser**
Adds a read option that passes the aggregate.

### CODE REVIEW

File: src/dproto_tcp.erl [hunk 1]
```diff
@@ -580,8 +683,23 @@ encode_r(R)
...
+    n;
 decode_r(?OPT_R_DEFAULT) ->
-    {r, default};
-decode_r(R) ->
-    {r, R}.
+    default;
+decode_r(R) when is_integer(R), R > 0 ->
+    R.
+
+-type aggr_bin() :: <<_:40,_:_*8>>.
```
Comment (elvisci) [inline src/dproto_tcp.erl]: Missing space right "," on line 692

**Comment by Licenser** [comment]
@davecromberge any thoughts? This only covers aggregates that are function(count) but I think that'll be enough for the start.

**Comment by davecromberge** [comment]
@Licenser  if I understand this correctly, this is the first step toward pushing some aggregation functions closer to the data, as a means of reducing the amount of data transferred between nodes?

I think it gets a little confusing in that someone could say {rr, on} and pass an aggregate.  It would make sense that aggregated results cannot be repaired, so that option should not be allowed.  Otherwise it looks good.

**Comment by Licenser** [comment]
Hmm good point but we could still read repair on errors, it would just cause another read, those could even scheudled for later on.

**Pull request merged**